jena-arq/src/main/java/org/apache/jena/sparql/serializer/QuerySerializer.java

```diff
@@ -142,7 +144,19 @@ public void visitAskResultForm(Query query)
         out.print("ASK") ;
         out.newline() ;
     }
-    
+
+    @Override
+    public void visitJsonResultForm(Query query) {
+        out.print("JSON {");
```

Comment:
This is used only when we create the `Query` manually. Only developers/advanced users might be interested in using it. The results via HTTP are not serialized this way.

Reply:
The `qparse` tool uses serialization (and it parses the result back again).

Reply:
More test suggestions: `testParse("JSON { "s": "FOO" , "p" : ?p , "o" : ?o } WHERE { }")` to parse to Q, serialize, reparse to Q2 and check Q.equals(Q2).

e.g.a new `org.apache.jena.sparql.syntax.TestJSONParsing`

`JSON { "s": "FOO" }` becomes `JSON { "s": FOO }`

Reply:
>JSON { "s": "FOO" } becomes JSON { "s": FOO }

Shouldn't it be

```
JSON { "s": "FOO" } becomes JSON { "s": "FOO" } // last FOO still with double quotes / a string?
```